storage/remote/read.go

```diff
@@ -151,11 +151,11 @@ type concreteSeriesSet struct {
 
 func (c *concreteSeriesSet) Next() bool {
 	c.cur++
-	return c.cur < len(c.series)
+	return c.cur-1 < len(c.series)
 }
 
 func (c *concreteSeriesSet) At() storage.Series {
-	return c.series[c.cur]
+	return c.series[c.cur-1]
```

Comment:
Intuitively I wouldn't expect `At` to return the series before the current position. Could you extend the test to include at least 2 series and show the behavior of Next and At there?

Reply:
Done

Reply:
Yep, a little counter-intuitive. This is what we do in tsdb: https://github.com/prometheus/tsdb/blob/dc8710380793a6d30b94e730edf6c8a4dae69945/postings.go#L319-L364

Reply:
Thanks @brian-brazil.